lib/Controller/DocumentController.php

```diff
@@ -871,50 +732,47 @@ private function getWopiInfoForPublicLink(array $docInfo) : array {
 	}
 
 	/**
-	 * @NoCSRFRequired
-	 * @PublicPage
-	 * Generates and returns an access token and urlsrc for a given fileId
-	 * for requests that provide secret token set in app settings
+	 * API endpoint to list all user documents
+	 *
+	 * lists the documents the user has access to (including shared files, once the code in core has been fixed)
+	 * also adds session and member info for these files
+	 *
+	 * @NoAdminRequired
 	 */
-	public function extAppWopiGetData($documentId) {
-		list($fileId, , $version, ) = Helper::parseDocumentId($documentId);
-
-		// If type of fileId is a string, then it
-		// doesn't work for shared documents, lets cast to int everytime
-		$fileId = (int)$fileId;
-
-		$secretToken = $this->request->getParam('secret_token');
-		$apps = \array_filter(\explode(',', $this->appConfig->getAppValue('external_apps')));
-		foreach ($apps as $app) {
-			if ($app !== '') {
-				if ($secretToken === $app) {
-					$appName = \explode(':', $app);
-					$this->logger->info('extAppWopiGetData(): External app "{extApp}" authenticated; issuing access token for fileId {fileId}', [
-						'app' => $this->appName,
-						'extApp' => $appName[0],
-						'fileId' => $fileId
-					]);
+	public function listAll() {
+		$rawDocuments = $this->documentService->getDocuments();
 
-					$retArray = [];
-					if ($doc = $this->getDocumentByUserAuth($this->uid, $fileId, null)) {
-						$retArray = $this->getWopiInfoForAuthUser($doc);
-						$retArray['urlsrc'] = $doc['urlsrc'];
-					}
+		$documents = [];
+		$locale = \strtolower(\str_replace('_', '-', $this->settings->getUserValue($this->uid, 'core', 'lang', 'en')));
+		foreach ($rawDocuments as $key=>$document) {
+			if (\is_object($document)) {
+				$documents[] = $document->getData();
+			} else {
+				$documents[$key] = $document;
+			}
 
-					return $retArray;
-				}
+			$wopiSrcUrl = $this->discoveryService->getWopiSrcUrl($document['mimetype']);
+			if (!$wopiSrcUrl) {
+				return [
+					'status' => 'error',
+					'message' => $this->l10n->t('Collabora Online: Unable to read WOPI discovery for given document', []),
+					'hint' => $this->l10n->t('Please contact the administrator.', [])
+				];
 			}
+
+			$documents[$key]['icon'] = \preg_replace('/\.png$/', '.svg', Template::mimetype_icon($document['mimetype']));
+			$documents[$key]['hasPreview'] = $this->previewManager->isMimeSupported($document['mimetype']);
+			$documents[$key]['urlsrc'] = $wopiSrcUrl['urlsrc'];
+			$documents[$key]['action'] = $wopiSrcUrl['action'];
+			$documents[$key]['locale'] = $locale;
 		}
 
-		return new JSONResponse([], Http::STATUS_UNAUTHORIZED);
-	}
+		\usort($documents, function ($a, $b) {
+			return @$b['mtime']-@$a['mtime'];
```

Comment:
we should include a comment here. I know we're suppressing the error message because we know the key might be missing, but taking into account this isn't common, I don't know what the value will be.
It's better to include some explanation about what's happening behind the scenes and what is the expected results. For example, I don't know if the documents without mtime will be first or last.

Reply:
very old code, lets not touch this. LEGACY tag is in the header.